### PR TITLE
Breaking changes to codegen

### PR DESCRIPTION
**Pull request by SferaDev**
- Make incremental builds the default setting
- Do not attempt to read the env variables by default
- Make the codegen more explicit
- Changes related to the new codegen file

### CODE REVIEW

File: packages/codegen/example/build-example.mjs [hunk 1]
```diff
@@ -21,7 +21,7 @@ async function main() {
     language: 'typescript',
     existingCode: undoReplaceImport(typescriptContents)
   });
-  const { javascript: mjs, types } = await generate({ schema, databaseURL, language: 'javascript' });
+  const { javascript: mjs, types } = await generate({ schema, databaseURL, language: 'javascript', });
```
Comment (SferaDev) [inline packages/codegen/example/build-example.mjs:24]:
```suggestion
  const { javascript: mjs, types } = await generate({ schema, databaseURL, language: 'javascript' });
```

File: packages/codegen/example/build-example.mjs [hunk 2]
```diff
@@ -31,11 +31,11 @@ async function main() {
 }
 
 function replaceImport(source) {
-  return source.replaceAll('@xata.io/client', '../../client/src');
+  return source?.replaceAll('@xata.io/client', '../../client/src');
```
Comment (eemmiillyy) [inline packages/codegen/example/build-example.mjs:34]: Is source possible undefined?

**Review by eemmiillyy — APPROVED** [review]
LGTM

File: packages/codegen/example/build-example.mjs (hunk 2 above)
Comment (SferaDev) [reply packages/codegen/example/build-example.mjs:34]: Yes, if the file doesn't exist previously

File: packages/codegen/src/codegen.ts [hunk 3]
```diff
@@ -361,3 +301,27 @@ function emitDeclarations(code: string) {
 
   return files.get('index.d.ts');
 }
+
```
Comment (richardgill) [inline packages/codegen/src/codegen.ts:304]: @SferaDev What's the plan for cloudflare workers?

**Comment by richardgill** [comment]
Looks like a good set of changes that moves us towards the notion doc 👍 🚀

**Review by richardgill — APPROVED** [review]

**Pull request merged**